src/algorithms/digi/SiliconChargeSharing.cc

```diff
@@ -74,6 +74,13 @@ void SiliconChargeSharing::process(const SiliconChargeSharing::Input& input,
                                       globalHitPos.z * dd4hep::mm),
                      transformIt->second);
 
+    // globalHitPos can sometimes lie outside of cell with cellID given by hit.getCellID()
+    // therefore, we search neighbors in a sensor to find the cell ID that correspond to globalHitPos
+    // Precise reason unknown, but we suspect it's cause by steps in Geant4
+    // Perhaps position is the average of all steps in volume while cellID is just the first cell the track hits
+    // They disagree when there are multiple step and scattering inside the volume
+    cellID = segmentationIt->second->cellID(hitPos, hitPos, cellID);
```

Comment:
```suggestion
    cellID = segmentationIt->second->cellID(hitPos, globalHitPos, cellID);
```
This one may be unused, but, I think, it would be a good practice to provide it correctly?

Reply:
What I meant by "not worth it" was "I am too lazy to type out a line of code for the conversion" :P, not any hit in the performance. 

Your way is better. The code is updated.